common/types/mapper/proto/shared_test.go

```diff
@@ -296,3 +300,60 @@ func TestCrossClusterApplyParentClosePolicyResponse(t *testing.T) {
 		),
 	)
 }
+
+func TestAny(t *testing.T) {
+	t.Run("sanity check", func(t *testing.T) {
+		internal := types.Any{
+			ValueType: "testing",
+			Value:     []byte(`test`),
+		}
+		rpc := sharedv1.Any{
+			ValueType: "testing",
+			Value:     []byte(`test`),
+		}
+		require.Equal(t, &rpc, FromAny(&internal))
+		require.Equal(t, &internal, ToAny(&rpc))
+	})
+
+	t.Run("round trip nils", func(t *testing.T) {
+		// somewhat annoying in fuzzing and there are few possibilities, so tested separately
+		assert.Nil(t, FromAny(ToAny(nil)), "nil proto -> internal -> proto => should result in nil")
+		assert.Nil(t, ToAny(FromAny(nil)), "nil internal -> proto -> internal => should result in nil")
+	})
+
+	t.Run("round trip from internal", func(t *testing.T) {
+		testutils.EnsureFuzzCoverage(t, []string{
+			"empty data", "filled data",
+		}, func(t *testing.T, f *fuzz.Fuzzer) string {
+			var orig types.Any
+			f.Fuzz(&orig)
+			out := ToAny(FromAny(&orig))
+			assert.Equal(t, &orig, out, "did not survive round-tripping")
+			// report what branch of behavior was fuzzed occurred
+			if len(orig.Value) == 0 {
+				return "empty data" // ignoring nil vs empty difference
+			}
+			return "filled data"
+		})
+	})
+	t.Run("round trip from proto", func(t *testing.T) {
+		testutils.EnsureFuzzCoverage(t, []string{
+			"empty data", "filled data",
```

Comment:
part of the reason I like having this "make sure you did actually do N branches" check:
it ensures that `f.Fuzz(&orig.ValueType)` actually does do different things, rather than "fuzzing" the same value over and over.  e.g. if it didn't mutate the data for some reason.